### DISCUSSION TITLE
Solver initialization

### DISCUSSION THREAD

**Discussion by richteropt** [casadi/casadi#3118] [General]
Hi!

If 'x0' in nlpsol is not defined, which initial primal iterate is chosen then: Is it set by casadi (e.g. as the zero vector) or is initialization left to the solver? The same question applies to the initial dual iterate.

Many thanks.

**Comment by brunoml5** [comment]
Hi! It appears that the default is set by casadi. The default value for bound is -inf/+inf and for the rest is zero.

See this, line 209
[https://github.com/casadi/casadi/blob/b88a5d63583114bc2abbc29857895fc146b503b3/casadi/core/nlpsol.cpp](url)

**Reply by richteropt** [reply]
Thanks! 

I think depending on the solver it would be an option to allow the solver to choose the initial point. E.g. Knitro has an option for IPM to choose an initial point based on some internal procedure. And this init point often is better than choosing the origin.
